backend/tracim_backend/lib/core/event.py

```diff
@@ -71,20 +74,61 @@ def __init__(
         self._session = session
         self._config = config
 
-    def get_messages_for_user(self, user_id: int, read_status: ReadStatus) -> typing.List[Message]:
-        query = (
-            self._session.query(Message)
-            .filter(Message.receiver_id == user_id)
-            .options(joinedload(Message.event))
-        )
+    def _base_query(
+        self,
+        read_status: ReadStatus = ReadStatus.ALL,
+        user_id: typing.Optional[int] = None,
+        event_id: typing.Optional[int] = None,
+    ) -> Query:
+        query = self._session.query(Message)
+        if event_id:
+            query = query.filter(Message.event_id == event_id)
+        if user_id:
+            query = query.filter(Message.receiver_id == user_id)
+        query = query.options(joinedload(Message.event))
         if read_status == ReadStatus.READ:
             query = query.filter(Message.read != null())
         elif read_status == ReadStatus.UNREAD:
             query = query.filter(Message.read == null())
         else:
             # ALL doesn't need any filtering an is the only other handled case
             assert read_status == ReadStatus.ALL
-        return query.all()
+        return query
+
+    def get_one_message(self, user_id: int, event_id: int) -> Message:
+        try:
+            return self._base_query(user_id=user_id, event_id=event_id).one()
+        except NoResultFound as exc:
+            raise MessageDoesNotExist(
+                'Message for user {} with event id "{}" not found in database'.format(
+                    user_id, event_id
+                )
+            ) from exc
+
+    def read_message_of_user(self, user_id, event_id) -> Message:
```

Comment:
With the method starting by `read`, I understand it reads the message with a "file read" semantic. Wouldn't `mark_message_of_user_as_read` be more explicit?
And given the order message_of_user, I would expect `(event_id, user_id)` for the order of arguments (so that it is the same as in the name)… But you have a `read_all(user_id)` method… Then use `mark_user_message_as_read(user_id, event_id)`?

Reply:
Fixed, better ?
I keep a same order everywhere with event_id, user_id which seems to me more clear.